Adding PottsCellFlyNeuron class and tests

Jannetty opened the pull request:
This is the simplest of the three agent classes I have implemented for the fly neurogenesis project.

**Estimated time to review**: Small - Medium

**Linked issue**: #50 , #91 

**Description of changes**:
- Adding PottsCellFlyNeuron class and tests

**Seeking feedback**:
Currently, I am relying on users to set the growth rate to 0 and the apoptosis rate to 0 in the setup file for this class to behave as I need it to. I recognize that I can get my needed functionality from this class by parameterize the "default" potts stem cell class, but the neurons are not stem cells so I think that is confusing, and the neuroblast cells which are the stem cells in this system have very different behavior. This class does not ensure that cells do not grow, but errors out if they try to divide or apoptose. This has been my working solution on my mess branch for a while, but open to alternative suggestions. If the optimal solution is to just appropriately parameterize the default potts stem cell class I will ask that we rename that class.

cainja commented:
@jessicasyu, does it make sense for `Parameters` to have setters for this type of behavior, so that these types of values can be set in the constructor? - I think this would be similar to how we handle stem cells, where the apoptosis age is set to a MAX_DOUBLE in the constructor.

jessicasyu commented:
To make sure I understand, the question we are trying to address is "how do we enforce that certain parameters are constant for a given cell type?" and the current options have been:

1. Assume the user sets them properly in the setup file (which we can't guarantee)
2. Set the parameter in the constructor (requires that the field is defined in the class)
3. Add setters to `Parameters` (I don't love the idea of introducing mutability)

I _think_ the most sustainable option would be to handle the parameters XML differently, so that each class has it's own (maybe inheritable) parameters file and we can tag some of them as constants. How do we feel about that?

cainja commented:
I think we are on the same page. 

1. I don't like this assumption and I think it makes ARCADE harder to use.
2. I think this issue is specific to requiring changes at the interface level, which I think scales poorly (I think the distinction would be somewhat arbitrary to a field or a parameter.
3. I go back and forth. 

I do think that a parameters XML change would be best and more sustainable (especially for documentation purposes) - but I also think designing and implementing that change could be a longer timeline because I can envision a lot of clunky ways of implementing. Let's create an issue to discuss/track? 

@Jannetty - As for the short-term/this PR, I think my opinion is that these checks can happen in the constructor until we can figure out a better solution (so that they fail on the initialization rather than when it becomes a problem).

Jannetty commented:
Thank you both for the helpful discussion. I made a very brief issue. I appreciate the advice for a shorter time horizon solution to this specific class I will move ahead with that

Jannetty commented:
I added some inelegant checks in the constructor to make sure the parameters are correctly set and adjusted the tests accordingly. I need to implement a make function to extend PottsCell so I left that exception in there even though, as currently written, it could never be called.

Jannetty commented:
@jessicasyu @cainja, everything above what I am about to push is what was discussed above. However, in reflecting on what I'm doing here, it also occurs to me that I think these cells should be quiescent, not proliferative with growth rate 0. I took a stab at implementing an extremely simple `PottsModuleQuiescenceSimple` that just doesn't do anything on a step and set the state module to Quiescent in the constructor.

I can alternatively set the module to null no matter what and add a check in the `PottsCell` step function so the module is only stepped if it is not null (this check exists in the `PatchCell` and this (I think) is how Quiescence is dealt with in the Patch cells).

Sorry for somewhat of a left turn here.

cainja commented:
I think it will be a little cleaner to use a higher level module such that those modules can be repeated for different cell types. Instead of adding `PottsModuleFlyNeuronQuiescence`, I think it would be slightly better to use the top level `PottsModuleQuiescence` and make the changes necessary to not throw an error when used with a different cell type.

jessicasyu commented on src/arcade/potts/agent/cell/PottsCellFlyNeuron.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Override
+    void setStateModule(CellState newState) {
+        if (!(newState instanceof State)) {
```
I'm not 100% sure, but I think you would just need to check if `State` is `null` and otherwise `CellState` in the constructor would guarantee that it is a `State`?

jessicasyu commented on src/arcade/potts/agent/cell/PottsCellContainer.java:
```diff
@@ -172,6 +172,8 @@ public Cell convert(
         // Make cell.
         switch (popParameters.get("CLASS")) {
+            case "fly-neuron":
+                return new PottsCellFlyNeuron(this, location, parameters);
```
I wonder if it makes more sense to keep `links` in the constructor for the neuron cells and pass `links` here, but keep the `null` usage in the `super` call. I think that the cell container class shouldn't necessarily care if a cell type does or does not need specific parameters (i.e. all cells have the same constructors from the view of the container class), but then the implementation can decide to use all those values or not. Maybe we could have the constructor log a warning that a specific user configuration is ignored (once we get logging figured out...)
